Makefile

```diff
@@ -14,7 +14,7 @@ PULUMI_CONVERT := 1
 
 # Override during CI using `make [TARGET] PROVIDER_VERSION=""` or by setting a PROVIDER_VERSION environment variable
 # Local & branch builds will just used this fixed default version unless specified
-PROVIDER_VERSION ?= 3.0.0-alpha.0+dev
+PROVIDER_VERSION ?= 4.0.0
```

Comment:
should this keep the -alpha suffix?

Reply:
It was the `?` that allowed the provider version action to override back to an alpa version (which resolves to v3 for module detection and thus the Go SDK build fails). I removed that for now.

We're considering improving https://github.com/pulumi/provider-version-action to correctly calculate a major version as well, which means we wouldn't have to set `VERSION_PREFIX` in `.ci-mgmt.yaml` anymore. For now, this should get overwritten in the nightly build.